src/NativeScript/inspector/SuppressAllPausesEnabler.cpp

```diff
@@ -0,0 +1,31 @@
+#include "SuppressAllPausesEnabler.h"
+#include <JavaScriptCore/Debugger.h>
+
+namespace NativeScript {
+
+SuppressAllPausesEnabler::SuppressAllPausesEnabler(JSC::JSGlobalObject& globalObject)
+    : m_globalObject(globalObject) {
+    if (this->suppressAllPauses()) {
+        if (JSC::Debugger* debugger = this->m_globalObject.debugger())
+            debugger->setSuppressAllPauses(true);
+    }
+}
+
+SuppressAllPausesEnabler::~SuppressAllPausesEnabler() {
+    if (this->suppressAllPauses()) {
+        if (JSC::Debugger* debugger = this->m_globalObject.debugger()) {
+            debugger->setSuppressAllPauses(false);
+        }
+    }
+}
+
+bool SuppressAllPausesEnabler::suppressAllPauses() {
+    static std::once_flag initializeEnvVariableOnceFlag;
+
+    std::call_once(initializeEnvVariableOnceFlag, [this]() {
+        this->m_dontSuppressAllPauses = getenv("NS_dontSuppressAllPauses");
```

Comment:
Let's make the prefix `TNS_` so that it matches the prefix of the runtime's public API.